backend/app/routes/admin/updateAdmin.js

```diff
@@ -4,33 +4,32 @@ const { ErrorHandler } = require('../../../helpers/error');
 const constants = require('../../../constants');
 
 module.exports = async (req, res, next) => {
-  const userRecord = await Admin.findOne({
-    email: res.locals.decode.email,
-  });
-
+  const { firstName, lastName, contact, username } = req.body;
   // Finding user through email which needs to be updated
-  const user = await Admin.findOne({ email: userRecord.email });
 
   // Checking which fields are present and making changes to the present field
-  if (user) {
-    user.username = req.body.username || user.username;
-    user.contact = req.body.contact || user.contact;
-    user.firstName = req.body.firstName || user.firstName;
-    user.lastName = req.body.lastName || user.Name;
-  }
 
   // Updating the document with updated data
-  const [err] = await to(Admin.findOneAndUpdate({ email: userRecord.email }, user));
+  // validation
+  const userbody = {
+    ...(firstName && { firstName }),
```

Comment:
why do you need this?

Reply:
Like the four fields in req.body are optional. So, to avoid setting up fields which haven't been mentioned in req.body to null, I used this.

Reply:
Find a neater way to handle this situation